gson/src/main/java/com/google/gson/internal/bind/ReflectiveTypeAdapterFactory.java

```diff
@@ -215,11 +221,24 @@ protected BoundField(String name, boolean serialized, boolean deserialized) {
         in.beginObject();
         while (in.hasNext()) {
           String name = in.nextName();
-          BoundField field = boundFields.get(name);
-          if (field == null || !field.deserialized) {
-            in.skipValue();
+          if (useJsog && JsogRegistry.JSOG_FIELD_ID.equals(name)) {
+            // JsogEnabled Id -- Registers the instance on JsogEnabled
+            String jsogId = in.nextString();
+            JsogRegistry.get().register(jsogId, instance);
+          } else if (useJsog && JsogRegistry.JSOG_FIELD_REF.equals(name)) {
+            // JsogEnabled backreference -- Replaces the instance with the value registered on JsogEnabled
```

Comment:
Maybe just write "JSOG ID..." "JSOG backreference..." since the annotation `@JsogEnabled` might not be used